main/src/test/java/org/mobilitydata/gtfsvalidator/validator/FeedExpirationDateValidatorTest.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2020 Google LLC, MobilityData IO
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.validator;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Locale;
+import org.junit.Test;
+import org.mobilitydata.gtfsvalidator.notice.FeedExpirationDateNotice;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.notice.ValidationNotice;
+import org.mobilitydata.gtfsvalidator.table.GtfsFeedInfo;
+import org.mobilitydata.gtfsvalidator.type.GtfsDate;
+
+public class FeedExpirationDateValidatorTest {
+
+  private List<ValidationNotice> validateFeedInfo(GtfsFeedInfo feedInfo) {
+    NoticeContainer container = new NoticeContainer();
+    FeedExpirationDateValidator validator = new FeedExpirationDateValidator();
+    validator.validate(feedInfo, container);
+    return container.getValidationNotices();
+  }
+
+  private GtfsFeedInfo createFeedInfo(GtfsDate feedEndDate) {
+    return new GtfsFeedInfo.Builder()
+        .setCsvRowNumber(1)
+        .setFeedPublisherName("feed publisher name value")
+        .setFeedPublisherUrl("www.mobilitydata.org")
+        .setFeedLang(Locale.CANADA)
+        .setFeedEndDate(feedEndDate)
+        .build();
+  }
+
+  @Test
+  public void feedExpiringInFiveDaysFromNowShouldGenerateNotice() {
+    assertThat(
+            validateFeedInfo(createFeedInfo(GtfsDate.fromLocalDate(LocalDate.now().plusDays(3)))))
+        .containsExactly(
+            new FeedExpirationDateNotice(
+                1,
+                GtfsDate.fromLocalDate(LocalDate.now()),
```

Comment:
Instead of using now and `plusDays()`, which mirrors the implementation, I would instead use exact dates. IMHO this way the test is always executing on the exact same dates, and the intent on whether the 7 and 30 days is inclusive or not is clearer. You're also more likely to catch bugs if you don't mirror the original implementation here.

Reply:
After discussion with @barbeau, we agreed on keeping the tests the way they are written because: `LocalDate.now()` is used in the use case implementation, and the current implementation does not seem to allow us to easily inject `GtfsDate` as a parameter of the use case. We will be revisiting this after the release of `v2.0`.